src/config.ts

```diff
@@ -24,6 +24,34 @@ export type CLSMechanism =
 
 export type ContextHeaderBehavior = 'default'|'ignore'|'require';
 
+export interface RequestDetails {
+  /**
+   * The request timestamp.
+   */
+  timestamp: number;
+  /**
+   * The request URL.
+   */
+  url: string;
+  /**
+   * The request method.
+   */
+  method: string;
+  /**
+   * The parsed trace context, if it exists.
+   */
+  traceContext: {traceId: string; spanId: string; options: number;}|null;
+  /**
+   * The original options object used to create the root span that corresponds
+   * to this request.
+   */
+  options: {};
+}
+
+export interface TracePolicy {
+  shouldTrace: (requestDetails: RequestDetails) => boolean;
```

Comment:
It seems like boilerplate to pass in an object wrapping a funciton. Could you just accept a function?

Reply:
I'd rather we accept the object. With propagation we would have another overriding class `Propagation` where it makes more sense to have an object, because there are three fields. This would also allow users to easily implement a class and pass in an instance of a class.